Sources/XMTPiOS/Libxmtp/DecodedMessageV2.swift

```diff
@@ -0,0 +1,506 @@
+        case .transactionReference:
+            return ContentTypeTransactionReference
+        case .groupUpdated:
+            return ContentTypeGroupUpdated
+        case .groupMembershipChanges:
+            return ContentTypeGroupMembershipChanges
+        case .readReceipt:
+            return ContentTypeReadReceipt
+        case .custom(let ffiEncodedContent):
+            if let typeId = ffiEncodedContent.typeId {
+                return ContentTypeID(
+                    authorityID: typeId.authorityId,
+                    typeID: typeId.typeId,
+                    versionMajor: Int(typeId.versionMajor),
+                    versionMinor: Int(typeId.versionMinor)
+                )
+            } else {
+                // Return a default content type if none is specified
+                return ContentTypeText
+            }
+        }
+    }
+
+    private func mapReaction(_ reactionPayload: FfiReactionPayload) -> Reaction {
+        return Reaction(
+            reference: reactionPayload.reference,
+            action: reactionPayload.action == .added ? .added : .removed,
+            content: reactionPayload.content,
+            schema: mapReactionSchema(reactionPayload.schema)
+        )
+    }
+
+    private func mapReactionSchema(_ schema: FfiReactionSchema) -> ReactionSchema {
+        switch schema {
+        case .unicode:
+            return .unicode
+        case .shortcode:
+            return .shortcode
+        case .custom:
+            return .custom
+        case .unknown:
+            return .unknown
+        }
+    }
+
+    private func mapRemoteAttachmentScheme(_ scheme: String) -> RemoteAttachment.Scheme {
+        return RemoteAttachment.Scheme(rawValue: scheme) ?? .https
+    }
+
+    private func mapFfiEncodedContent(_ ffiContent: FfiEncodedContent) throws -> EncodedContent {
+        var encoded = EncodedContent()
+        if let typeId = ffiContent.typeId {
+            encoded.type = ContentTypeID(
+                authorityID: typeId.authorityId,
+                typeID: typeId.typeId,
+                versionMajor: Int(typeId.versionMajor),
+                versionMinor: Int(typeId.versionMinor)
+            )
+        }
+        encoded.parameters = ffiContent.parameters
+        encoded.fallback = ffiContent.fallback ?? ""
+        encoded.content = ffiContent.content
+        if let compression = ffiContent.compression {
+            // Map Int32 compression values to the enum
+            // Assuming 0 = deflate, 1 = gzip based on the proto definition
+            switch compression {
+            case 0:
+                encoded.compression = .deflate
+            case 1:
+                encoded.compression = .gzip
+            default:
+                // Unknown compression value, leave it unset
+                break
+            }
+        }
+        return encoded
+    }
+}
+
+// I included this for completeness in Rust, but I think it's unused except for really old groups.
```

Comment:
Should I just nuke this from Rust? I added it for completeness, but isn't it kinda dead and replaced with `GroupUpdated`? Would only be there for legacy reasons.